contracts/nft/distributor/entry/claim.sol

```diff
@@ -42,8 +47,32 @@ contract ent_claim_NFTDistributor_v1 is
             for (uint256 j; j < requests[i].amount; j++) {
                 distributorStore().nft.mint(tierIndex, account);
             }
+
+            uint256[] memory postOwnedTokens =
+                distributorStore().nft.getOwnedTokens(account);
+
+            uint256[] memory newlyMintedTokenIds =
+                new uint256[](requests[i].amount);
+
+            uint256 offset = postOwnedTokens.length - requests[i].amount;
+
+            for (uint256 k; k < requests[i].amount; k++) {
+                newlyMintedTokenIds[k] = postOwnedTokens[k + offset];
+            }
+
+            distributorStore().dictionary.setTokenTierForTokenIds(
+                newlyMintedTokenIds,
+                tierIndex
+            );
```

Comment:
adding inline comments here would be super beneficial when we come back to this at a later point